### PR TITLE
install.sh: fail on download errors

### LINKED ISSUE #5145: Install script fails with "End-of-central-directory signature not found."
#### What is the problem you are having with rclone?
Running the install script results in an error

```
Archive:  rclone-current-osx-arm.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of rclone-current-osx-arm.zip or
        rclone-current-osx-arm.zip.zip, and cannot find rclone-current-osx-arm.zip.ZIP, period.
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

macOS Big Sur 11.2.3

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

curl https://rclone.org/install.sh | sudo bash

or

curl https://rclone.org/install.sh | sudo bash -s beta

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
curl https://rclone.org/install.sh | sudo bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  4436  100  4436    0     0  27214      0 --:--:-- --:--:-- --:--:-- 27048
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    15  100    15    0     0     65      0 --:--:-- --:--:-- --:--:--    65
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
Archive:  rclone-current-osx-arm.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of rclone-current-osx-arm.zip or
        rclone-current-osx-arm.zip.zip, and cannot find rclone-current-osx-arm.zip.ZIP, period.
```

### PR DESCRIPTION
**Pull request by ivandeex**
#### What is the purpose of this change?

Before this change if the download script was run on MacOS ARM, it attempted
to request the "current-osx-arm.zip" archive and unzip it, resulting in
error `cannot find zipfile directory` because we don't provide such a build.

**EDIT: for v1.55 we will support osx/arm**

This patch makes `install.sh`:
- ~~request amd64 build on MacOS arm~~
- always run curl with flag `-f` (aka `--fail`) so it fails explicitly on any `40x`/`50x` http status

Additionally it fixes few cosmetic _shellcheck_ warnings:
- replaces obsolete backticks with `$(...)`
- quotes bare variables

#### Was the change discussed in an issue or in the forum before?

Fixes #5145

#### Checklist

- [x] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [ ] I have added tests for all changes in this PR if appropriate.
- [ ] I have added documentation for the changes if appropriate.
- [x] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [x] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: docs/content/install.sh [hunk 1]
```diff
@@ -113,45 +113,48 @@ case $OS_type in
     ;;
 esac
 
+if [ "$OS" = 'osx' ] && [ "$OS_type" = 'arm' ]; then
```
Comment (ncw) [inline docs/content/install.sh:116]: This is OK for the moment but for v1.55 we will support osx/arm

**Review by ncw — COMMENTED** [review]
This maybe should be 2 commits - one cleanup and one the actual change.

I'm not sure if we should merge it though as we'll want to revert it for v1.55 which does support osx/arm64

Maybe we should merge the osx bit to the stable branch only and I can push that up to the website immediately. It will then get overwritten when we release v1.55

File: docs/content/install.sh (hunk 1 above)
Comment (ivandeex) [reply docs/content/install.sh:116]: Let's keep only the "fail on download errors" logic in the PR.
You can add the "arm -> amd64" trick on the stable branch directly.

**Comment by ivandeex** [comment]
> This maybe should be 2 commits - one cleanup and one the actual change.

Done

> I'm not sure if we should merge it though as we'll want to revert it for v1.55 which does support osx/arm64

I removed the `arm -> amd64` bit from the patch.

> Maybe we should merge the osx bit to the stable branch only and I can push that up to the website immediately. It will then get overwritten when we release v1.55

We can add that bit on `v1.54-stable` directly.

**Review by ncw — APPROVED** [review]
That looks great now - thanks

Please merge :-)

**Pull request merged**